client/src/app/site/deployment-center/deployment-center-setup/step-configure/configure-vsts-build/functions-frameworks/functions-frameworks.component.ts

```diff
@@ -29,6 +30,21 @@ export class FunctionsFramworksComponent {
       value: WebAppFramework.PrecompiledFunction,
     },
   ];
+
+  private setupValidators() {
+    const workingDirectoryValidator = RegexValidator.create(
+      new RegExp(/^(\.{2,}|~|())\\(.)*$|^(\.{2,}|~|())\/(.)*$|(.)+:(.)*/),
+      this._translateService.instant(PortalResources.validate_workingDirectory),
+      true
+    );
+    this.wizard.buildSettings.get('applicationFramework').valueChanges.subscribe(stack => {
```

Comment:
should we use takeuntil(this._ngUnsubscribe$) for subscription to  be unsubscribed when not needed.

Reply:
Done